frontend/src/providers/language/LanguageProvider.tsx

```diff
@@ -0,0 +1,32 @@
+import { createContext, FC, PropsWithChildren, useContext } from 'react';
+import {
+  PersistentStateKey,
+  usePersistentState,
+} from '../../state/persistentState';
+import { LanguageValue, LANGUAGES } from './constants';
+
+type LanguageContextType = {
+  language: string;
+  changeInAppLanguage: (newLanguage: LanguageValue) => void;
+};
+
+const LanguageContext = createContext<LanguageContextType>(null!);
+
+export const LanguageProvider: FC<PropsWithChildren> = ({ children }) => {
+  const [language, setLanguage] = usePersistentState(
```

Comment:
No need for Context/Provider - usePersistentState() handles the state. Check other examples of this hook usage in the app.